packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -110,14 +111,10 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
   const isWorker = config.isWorker
   const insertPreload = !(ssr || !!config.build.lib || isWorker)
 
-  const relativePreloadUrls = config.base === './' || config.base === ''
-
   const scriptRel = config.build.polyfillModulePreload
     ? `'modulepreload'`
     : `(${detectScriptRel.toString()})()`
-  const assetsURL = relativePreloadUrls
-    ? `function(dep,importerUrl) { return new URL(dep, importerUrl).href }`
-    : `function(dep) { return ${JSON.stringify(config.base)}+dep }`
+  const assetsURL = `function(dep,importerUrl) { return new URL(dep, importerUrl).href }`
```

Comment:
Same here, let's keep the code as it was before so we generate `/{base}/...` URLs if the base isn't relative.